requirements.txt

```diff
@@ -1,23 +1,171 @@
-boto3
-kombu==4.6.3 # CeleryExecutor issues with 1.10.2 supposedly fixed in 1.10.5 airflow, but still observed issues on 1.10.7
-apache-airflow[celery,postgres,hive,hdfs,jdbc,async,password,crypto,github_enterprise,datadog,statsd,s3,mysql,google_auth,gcp_api,kubernetes]==1.10.7
-mozlogging
-retrying
-newrelic
-redis
-hiredis
-requests
-jsonschema
-Flask-OAuthlib
-pytz
-werkzeug==0.16.0
-# The next requirements are for kubernetes-client/python
-urllib3>=1.24.2 # MIT
-ipaddress>=1.0.17;python_version=="2.7"  # PSF
-websocket-client>=0.32.0,!=0.40.0,!=0.41.*,!=0.42.* # LGPLv2+
-# Pin to older version, newer version has issues
-JPype1==0.7.1
-shelljob==0.5.6
-# Fix no inspection available issue
-# https://github.com/apache/airflow/issues/8211
-SQLAlchemy==1.3.15
+#
+# This file is autogenerated by pip-compile
+# To update, run:
+#
+#    pip-compile
```

Comment:
`pip-compile` will generate a list of dependencies that is specific to the python version installed, so I would recommend adding a job to CI enforcing that `requirement.txt` was generated from the correct python version, e.g. https://github.com/mozilla/bigquery-etl/blob/master/.circleci/config.yml#L42-L46

Reply:
Good suggestion, thanks!